### PR TITLE
feat(axelar-gateway): add golden test for message approval hash

### PR DESCRIPTION
**Pull request by ahramy**
[AXE-6123](https://axelarnetwork.atlassian.net/browse/AXE-6123)

[AXE-6123]: https://axelarnetwork.atlassian.net/browse/AXE-6123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/axelar-soroban-interfaces/tests/types.rs [hunk 1]
```diff
@@ -59,3 +59,41 @@ fn weighted_signers_hash() {
...
+        hex!("cfa347779c9b646ddf628c4da721976ceb998f1ab2c097b52e66a575c3975a6c"),
+        hex!("fb5eb8245e3b8eb9d44f228ee142a3378f57d49fc95fa78d437ff8aa5dd564ba"),
+        hex!("90e3761c0794fbbd8b563a0d05d83395e7f88f64f30eebb7c5533329f6653e84"),
+        hex!("60e146cb9c548ba6e614a87910d8172c9d21279a3f8f4da256ff36e15b80ea30"),
+    ]
+    .map(|hash| BytesN::<32>::from_array(&env, &hash));
+
+    let mut messages_array = soroban_sdk::Vec::new(&env);
+
+    for (i, payload_hash) in payload_hashes.iter().enumerate() {
```
Comment (milapsheth) [inline contracts/axelar-soroban-interfaces/tests/types.rs:77]:
```suggestion
    for (i, payload_hash) in payload_hashes.into_iter().enumerate() {
```

File: contracts/axelar-soroban-interfaces/tests/types.rs [hunk 2]
```diff
@@ -59,3 +59,41 @@ fn weighted_signers_hash() {
...
+            message_id: String::from_str(&env, &format!("test-{}", i + 1)),
+            source_address: String::from_str(
+                &env,
+                "CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHK3M",
+            ),
+            contract_address: Address::from_string(&String::from_str(
+                &env,
+                "CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDR4",
+            )),
+            payload_hash: payload_hash.clone(),
```
Comment (milapsheth) [inline contracts/axelar-soroban-interfaces/tests/types.rs:89]:
```suggestion
            payload_hash,
```
since we get the value instead of a reference

**Review by milapsheth — APPROVED** [review]

File: contracts/axelar-soroban-interfaces/tests/types.rs [hunk 3]
```diff
@@ -59,3 +59,41 @@ fn weighted_signers_hash() {
...
+        hex!("fb5eb8245e3b8eb9d44f228ee142a3378f57d49fc95fa78d437ff8aa5dd564ba"),
+        hex!("90e3761c0794fbbd8b563a0d05d83395e7f88f64f30eebb7c5533329f6653e84"),
+        hex!("60e146cb9c548ba6e614a87910d8172c9d21279a3f8f4da256ff36e15b80ea30"),
+    ]
+    .map(|hash| BytesN::<32>::from_array(&env, &hash));
+
+    let mut messages_array = soroban_sdk::Vec::new(&env);
+
+    for (i, payload_hash) in payload_hashes.iter().enumerate() {
+        messages_array.push_back(Message {
```
Comment (milapsheth) [inline contracts/axelar-soroban-interfaces/tests/types.rs:78]:
```suggestion
        messages.push_back(Message {
```

File: contracts/axelar-soroban-interfaces/tests/types.rs [hunk 4]
```diff
@@ -59,3 +59,41 @@ fn weighted_signers_hash() {
...
+                &env,
+                "CAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMDR4",
+            )),
+            payload_hash: payload_hash.clone(),
+        });
+    }
+
+    let approval_hash = env
+        .crypto()
+        .keccak256(&(CommandType::ApproveMessages, messages_array).to_xdr(&env))
```
Comment (milapsheth) [inline contracts/axelar-soroban-interfaces/tests/types.rs:95]:
```suggestion
        .keccak256(&(CommandType::ApproveMessages, messages).to_xdr(&env))
```

File: contracts/axelar-soroban-interfaces/tests/types.rs [hunk 5]
```diff
@@ -59,3 +59,41 @@ fn weighted_signers_hash() {
...
+
+    let payload_hashes = [
+        hex!("cfa347779c9b646ddf628c4da721976ceb998f1ab2c097b52e66a575c3975a6c"),
+        hex!("fb5eb8245e3b8eb9d44f228ee142a3378f57d49fc95fa78d437ff8aa5dd564ba"),
+        hex!("90e3761c0794fbbd8b563a0d05d83395e7f88f64f30eebb7c5533329f6653e84"),
+        hex!("60e146cb9c548ba6e614a87910d8172c9d21279a3f8f4da256ff36e15b80ea30"),
+    ]
+    .map(|hash| BytesN::<32>::from_array(&env, &hash));
+
+    let mut messages_array = soroban_sdk::Vec::new(&env);
```
[the remark this answers is not in the document]
Comment (ahramy) [reply contracts/axelar-soroban-interfaces/tests/types.rs:75]: updated.

File: contracts/axelar-soroban-interfaces/tests/types.rs (hunk 1 above)
Comment (ahramy) [reply contracts/axelar-soroban-interfaces/tests/types.rs:77]: updated.

File: contracts/axelar-soroban-interfaces/tests/types.rs (hunk 2 above)
Comment (ahramy) [reply contracts/axelar-soroban-interfaces/tests/types.rs:89]: good call. updated.

File: contracts/axelar-soroban-interfaces/tests/types.rs (hunk 4 above)
Comment (ahramy) [reply contracts/axelar-soroban-interfaces/tests/types.rs:95]: updated

**Pull request merged**